lib/dor/etd/identity_metadata_generator.rb

```diff
@@ -53,17 +57,16 @@ def generate
               unless old_identity_doc.nil?
                 catkey = old_identity_doc.at_xpath('//catkey')
                 catkey = old_identity_doc.at_xpath("//otherId[@name='catkey']") if catkey.nil?
-                xml.text(catkey.text)
+                xml.text(catkey.text) if catkey
```

Comment:
look!  this was a bug I squashed after my tests found it!

Reply:
The  ETDs are always going to have a catkey, so maybe we don't want a conditional.

Reply:
I wonder if there are any extant ETDs that aren't past this step that either don't have a catkey, or have an empty one.   If both of these are errors, there might be a better way to surface such a problem beyond "no method "text" on nil object".

I defer to your judgement -- it's just that in general, our metadata is in all sorts of states as requirements have changed over time.  For ETDs, I imagine there aren't a lot of old ones stuck in the middle of their accessioning.

We do have SOME, tho.  Perhaps @andrewjbtw can weigh in.

![image](https://user-images.githubusercontent.com/96775/75299052-ec7f0d00-57e8-11ea-8522-695862d76411.png)